### PR TITLE
Collapse UnicodeCharBuffer into CharBuffer

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Collapse UnicodeCharBuffer into CharBuffer

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2604)

### CODE REVIEW

File: src/Common/src/UnicodeCharBuffer.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace System.Windows.Forms
+{
+    public sealed class UnicodeCharBuffer
```
Comment (RussKie) [inline src/Common/src/UnicodeCharBuffer.cs]: ❗️ Why is this `public` instead of `internal`?

File: src/Common/src/UnicodeCharBuffer.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        internal char[] buffer;
+        internal int offset;
```
Comment (RussKie) [inline src/Common/src/UnicodeCharBuffer.cs]: I don't believe these are used outside the class

**Comment by RussKie** [comment]
Do you think it can be replaced with a `Span<char>`?

File: src/Common/src/UnicodeCharBuffer.cs (hunk 1 above)
Comment (hughbe) [reply src/Common/src/UnicodeCharBuffer.cs]: fixed

File: src/Common/src/UnicodeCharBuffer.cs (hunk 2 above)
Comment (hughbe) [reply src/Common/src/UnicodeCharBuffer.cs]: fixed

**Comment by hughbe** [comment]
> Do you think it can be replaced with a Span<char>?

I don't think so, mainly because `FileDialog` stores `_charBuffer` in a field (not supported by spans), because it uses `_charBuffer` in `HookProc`

**Pull request merged**